Gems/ROS2/Code/Tests/SdfParserTest.cpp

```diff
@@ -150,16 +134,95 @@ namespace UnitTest
                         </model>
                       </sdf>)";
         }
+
+        static std::string GetSdfWithDuplicateModelName()
+        {
+          return R"(<?xml version="1.0"?>
+            <sdf version="1.7">
+            <model name="root_model">
+              <link name="root_link"/>
+            </model>
+            <world name="default">
+              <model name="my_model">
+                <link name="link1"/>
+              </model>
+              <model name="your_model">
+                <link name="link2"/>
+              </model>
+              <model name="my_model">
+                <link name="link3"/>
+              </model>
+            </world>
+          </sdf>)";
+        }
+
+        static std::string GetSdfWithWorldThatHasMultipleModels()
+        {
+          return R"(<?xml version="1.0"?>
+            <sdf version="1.8">
+            <model name="root_model">
+              <link name="root_link"/>
+            </model>
+            <world name="default">
+              <model name="my_model">
+                <link name="link1"/>
+              </model>
+              <model name="your_model">
+                <link name="link2"/>
+              </model>
+            </world>
```

Comment:
My interpretation of the standard is that underneath the sdf tag, you can have a model, one or more worlds, a light, OR an actor, but not a combination of them:

http://sdformat.org/spec
<sdf> Description: SDFormat base element that can include one model, actor, light, or worlds. A user of multiple worlds could run parallel instances of simulation, or offer selection of a world at runtime.

Reply:
Well that is what I need to test.
It was also my understanding that either a single `<model>` tag or `<world>` tag at the root would be supported.
The `<world>` tag can support multiple `<model>` tag.

Once I can get my environment fixed, I can see if there any any errors in the above scenario.

It would be better if the http://sdformat.org/spec posted an xsd file that showed whether having a both a `<model>` tag and a `<world>` tag at the root is supported.

The statement on the page says
> SDFormat base element that can include one model, actor, light, or worlds

But is the "or" in that statement exclusive or does it mean "and/or"

Reply:
Actually it turns out that a combination of support a `<model>` tag at the root with multiple `<world>` tags at the root does parse successfully in the SDF.

The [SDFParserTest.SdfWithModelsOnRootAndWorld_ParsesSuccessfully](https://github.com/o3de/o3de-extras/pull/516/files#diff-e72d33e5f1cb617c5d05af4b8600545c3f20ece6c6d1aedc1bc2f1ff69e37f5bR194) validates that an SDF successfully parses when both a `<model>` and multiple `<world>` tags exist

Reply:
Huh, ok then. :) Thanks for testing it!